lg_media/scripts/image_viewer.py

```diff
@@ -0,0 +1,149 @@
+            rospy.logwarn('Director message did not contain valid data')
+            return
+        except ValueError:
+            rospy.logwarn('Director message did not contain valid json')
+            return
+        except TypeError:
+            rospy.logwarn('Director message did not contai valid type. Type was %s, and content was: %s' % (type(message), message))
+            return
+        for window in message.get('windows', []):
+            if window.get('activity', '') == 'image':
+                image = ImageView()
+                image.url = window['assets'][0]
+                image.geometry = WindowGeometry(
+                    width=window['width'],
+                    height=window['height'],
+                    x=window['x_coord'],
+                    y=window['y_coord']
+                )
+                image.viewport = window['presentation_viewport']
+                offset_geometry = ManagedWindow.lookup_viewport_geometry(image.viewport)
+                image.geometry.x = image.geometry.x + offset_geometry.x
+                image.geometry.y = image.geometry.y + offset_geometry.y
+                image.uuid = str(uuid.uuid4())
+                windows_to_add.images.append(image)
+        self.image_view_pub.publish(windows_to_add)
+
+    def is_in_current_images(self, current_images, image):
+        for _image, _image_obj in current_images.items():
+            if _image.url == image.url and \
+                    _image.geometry == image.geometry:
+                return _image_obj
+        return None
+
+    def handle_image_views(self, msg):
+        with self.lock:
+            self._handle_image_views(msg)
+
+    def _handle_image_views(self, msg):
+        new_current_images = {}
+        images_to_remove = self.current_images.values()
+        images_to_add = []
+        for image in msg.images:
+            # rospy.logerr('CURRENT IMAGES: {}\n\n'.format(self.current_images))
+            if image.viewport not in self.viewports:
+                continue
+            duplicate_image = self.is_in_current_images(self.current_images, image)
+            if duplicate_image:
+                # rospy.logerr('Keeping image: {}\n\n'.format(image))
+                images_to_remove.remove(duplicate_image)
+                new_current_images[image] = duplicate_image
+                continue
+            rospy.logwarn('Appending IMAGE: {}\n\n'.format(image))
+            images_to_add.append(image)
+
+        for image_obj in images_to_remove:
+            image_obj.set_state(ApplicationState.STOPPED)
+            file_to_remove = image_obj.cmd[-1]
+            if os.path.exists(file_to_remove):
+                os.remove(file_to_remove)
+        #images_to_remove = []
+        for image in images_to_add:
+            new_current_images[image] = self._create_image(image)
+
+        self.current_images = new_current_images
+        print 'current images {}'.format(self.current_images)
```

Comment:
Since Python 3 porting is incoming, good to use the new `print()` and import the usual `__future__`:

```python
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals
```